pkg/image/api/helper.go

```diff
@@ -1001,3 +1002,57 @@ func SetContainerImageEntrypointAnnotation(annotations map[string]string, contai
 func LabelForStream(stream *ImageStream) string {
 	return fmt.Sprintf("%s/%s", stream.Namespace, stream.Name)
 }
+
+// JoinImageSignatureName joins image name and custom signature name into one string with # separator.
+func JoinImageSignatureName(imageName, signatureName string) (string, error) {
+	if len(imageName) == 0 {
+		return "", fmt.Errorf("imageName may not be empty")
+	}
+	if len(signatureName) == 0 {
+		return "", fmt.Errorf("signatureName may not be empty")
+	}
+	if strings.Count(imageName, "#") > 0 || strings.Count(signatureName, "#") > 0 {
+		return "", fmt.Errorf("neither imageName nor signatureName can contain '#'")
+	}
+	return fmt.Sprintf("%s#%s", imageName, signatureName), nil
+}
+
+// SplitImageSignatureName splits given signature name into image name and signature name.
+func SplitImageSignatureName(imageSignatureName string) (imageName, signatureName string, err error) {
+	segments := strings.Split(imageSignatureName, "#")
+	switch len(segments) {
+	case 2:
+		signatureName = segments[1]
+		fallthrough
+	case 1:
```

Comment:
No need for this check. If you hit this directly, you were returning an error.  Just make the `default` case say that `imageName#imageSignature` is the required format.